theory-test-admission-candidate-app/app/server/services/candidateCheckBookingService.js

```diff
@@ -1,21 +1,61 @@
 const AWS = require('aws-sdk');
+const moment = require('moment');
+const logger = require('logger');
+
+const bookingStepFunction = new AWS.StepFunctions();
 
 export default class CandidateCheckBookingService {
 
-	checkCandidateBooking() {
-
-		return true;
-		// const bookingStepFunction = new AWS.StepFunctions();
-		// const params = {
-		// 	stateMachineArn: process.env.SFN_START_CANDIDATE_ADMISSION_ARN,
-		// 	input: { DrivingLicenceNumber: this.candidateDLN, AdmissionId: this.admissionID, ReceivedDate: new Date() }
-		// };
-		// bookingStepFunction.startExecution(params, (err, data) => {
-		// 	if (err) {
-		// 		return false; // an error occurred
-		// 	}
-		// 	return data.HasBooking; // successful response
-		// });
+	static checkCandidateBooking(admissionId, candidateDLN, callback) {
+
+		logger.info('Check candidate booking executed');
+		const currentDate = moment(Number.parseInt(new Date().getTime(), 10)).toISOString(true);
+
+		const dln = `"DrivingLicenceNumber": "${candidateDLN}"`;
+		const aId = `"AdmissionId": "${admissionId}"`;
+		const date = `"Date": "${currentDate}"`;
+		const completeInput = `{ "Request": {${dln}, ${aId}, ${date}} }`;
+
+		const params = {
+			stateMachineArn: process.env.SFN_START_CANDIDATE_ADMISSION_ARN,
+			input: completeInput
+		};
+		this.doStartExecution(params, (response) => {
+			logger.info('Start execution executed and returned: ', response);
+			const descParams = {
+				executionArn: response.executionArn,
+			};
 
+			new Promise((r, j) => {
+				this.doDescribeExecution(r, j, descParams);
+			}).then((result) => {
+				const jsonResponse = JSON.parse(result.output);
+				return callback(jsonResponse.HasBooking);
+			});
+
+		});
+	}
+	static doStartExecution(params, callback) {
+		bookingStepFunction.startExecution(params, (error, data) => {
+			if (error) {
+				logger.error('Start execution failed with the following error: ', error);
+			}
+			logger.info('Start execution succeeded with the following response: ', data);
+			return callback(data);
+		});
+	}
+	static doDescribeExecution(resolve, reject, params) {
+		bookingStepFunction.describeExecution(params, (error, response) => {
+			if (error) {
+				logger.error('Describe execution failed with the following error: ', error);
+				reject(error); // reject the promise
+			} else if (response.status !== 'SUCCEEDED') {
+				logger.info('Describe execution succeeded with the following response: ', response.status);
+				this.doDescribeExecution(resolve, null, params); // Try again
```

Comment:
There's no sleep here, you're polling AWS as fast as they'll reply.

Reply:
Done